### DISCUSSION TITLE
Same transaction with two records in CSV

### DISCUSSION THREAD

**Discussion by Moc** [firefly-iii/firefly-iii#7797] [Firefly III]
**Context:**
- Bank: ING 
- One private bank account (Name: `PRIVATE_NAME`, IBAN: `PRIVATE_IBAN` ). Entered as "default asset account" in Firefly. 
- One shared bank account with my partner (Name: `SHARED_NAME`, IBAN, `SHARED_IBAN`). Entered as "shared asset account" in Firefly. 

When I export my transactions from ING, it produces the following lines. 

**CSV file:**

| Datum  | Naam / Omschrijving | Rekening | Tegenrekening | Code | Af Bij | Bedrag (EUR) | Mutatiesoort | Mededelingen |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| 20220325  | `PRIVATE_NAME` | `SHARED_IBAN` | `PRIVATE_IBAN` | GT | Bij | 1000 | Online bankieren | Naam: `PRIVATE_NAME` Omschrijving: `DESCRIPTION` IBAN: `PRIVATE_IBAN` Datum/Tijd: 25-03-2022 08:05:49 Valutadatum: 25-03-2022	|
| 20220325  | `SHARED_NAME` | `PRIVATE_IBAN` | `SHARED_IBAN` | GT | Af | 1000 | Online bankieren | Naam: `SHARED_NAME` Omschrijving: `DESCRIPTION` IBAN: `SHARED_IBAN` Datum/Tijd: 25-03-2022 08:05:49 Valutadatum: 25-03-2022 |

**Problem:**
The CSV creates two records for what is essentially the same transaction. One from the point of view from my private bank account, and one from the point of view from the bank account that I share with my partner. Understandably, the Data Importer imports them as two separate transactions. This does however mess up my balances, as my private account balance is now `-2000` (2x `1000`) rather than `-1000`. 

This situation occurs every month in our case, as I transfer an X amount from my private account to the shared account monthly. 

**Question:**
What would be the best way to solve this problem?

**Comment by JC5** [comment]
ING often adds a transaction ID which is different for both transactions or the descriptions are different. That usually messes up the detection.

**Reply by Moc** [reply]
That makes sense of course :) But how do I best solve this? Is manually checking the CSV and removing the duplicate lines the only solution?

**Reply by JC5** [reply]
If all data is the same, then Firefly III will detect them, but that's not often the case.

**Reply by Moc** [reply]
So I tried Nordigen importing but that resulted in exactly the same duplicated transaction (as well as additional `<br>` tags that I don't want) so I decided to manually clean out the CSV file before importing. 

In this case I used the following Excel formula to help me detect possible duplicates (along with some conditional formatting to highlight the matches: 
```
=ÌF(OR(AND(C2="PRIVATE_IBA`";D2="SHARED_IBAN");AND(C2="SHARED_IBAN";D2="PRIVATE_IBAN"));"POSSIBLE PROBLEM";"")
```
